app/controllers/test/PartnershipIdentificationStubController.scala

```diff
@@ -32,22 +33,51 @@ class PartnershipIdentificationStubController @Inject()(mcc: MessagesControllerC
 
   def createJourney(partyType: String): Action[PartnershipIdJourneyConfig] = Action(parse.json[PartnershipIdJourneyConfig]) {
     journeyConfig =>
-      val journeyId = if (journeyConfig.body.businessVerificationCheck) partyType + excludeBv else partyType
+      val journeyId = if (journeyConfig.body.businessVerificationCheck) s"$partyType:$excludeBv" else partyType
 
       Created(Json.obj("journeyStartUrl" -> JsString(journeyConfig.body.continueUrl + s"?journeyId=$journeyId")))
   }
 
   def retrieveValidationResult(journeyId: String): Action[AnyContent] = Action.async {
     Future.successful(
-      Ok(Json.toJson(PartnershipIdEntity(
-        sautr = Some("1234567890"),
-        postCode = if (journeyId.equals("1")) Some("AA11AA") else None,
-        registration = "REGISTERED",
-        businessVerification = if (journeyId.contains(excludeBv)) None else Some(BvPass),
-        bpSafeId = Some("testBpId"),
-        identifiersMatch = true
-      ))(PartnershipIdEntity.apiFormat))
+      Ok(partnershipResponse(journeyId))
     )
   }
 
+  private def partnershipResponse(journeyId: String): JsValue = {
+    val partnershipEntity: JsValue = Json.toJson(PartnershipIdEntity(
+      sautr = Some("1234567890"),
+      postCode = if (journeyId.equals("1")) Some("AA11AA") else None,
+      registration = "REGISTERED",
+      businessVerification = if (journeyId.contains(excludeBv)) None else Some(BvPass),
+      bpSafeId = Some("testBpId"),
+      identifiersMatch = true
+    ))(PartnershipIdEntity.apiFormat)
+
+    val partyType: PartyType = PartyType.fromString(journeyId.split(":").head)
+    partyType match {
+      case LtdPartnership | ScotLtdPartnership | LtdLiabilityPartnership =>
+        val companyProfile = Json.parse("""
```

Comment:
Just from a style point of view it'd be nice if this was Json.obj() instead of Json.parse() are those are more readable and much easier to update if needed

Reply:
👍